src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx

```diff
@@ -133,9 +316,74 @@ class WidgetCard extends React.Component<Props> {
       );
     }
 
+    if (errorMessage) {
+      return (
+        <ErrorPanel>
+          <IconWarning color="gray500" size="lg" />
+        </ErrorPanel>
+      );
+    }
+
     const {location, router, selection} = this.props;
     const {start, end, period} = selection.datetime;
 
+    if (widget.displayType === 'world_map') {
+      const DEFAULT_GEO_DATA = {
+        title: '',
+        data: [],
+      };
+
+      const processTableResults = () => {
```

Comment:
Should this be done in `WidgetQueries` it looks like you have a path there for worldmaps and having the request and result processing steps co-located would be good. That would leave this component focused on selecting the right rendering